### ISSUE TITLE
"Rule-based" baselines

### ISSUE THREAD

**Issue by JamesOwers** [JamesOwers/midi_degradation_toolkit#71]
Currently we don't know how good our baselines are against super dumb baselines:
* Task 1: [1/9, 8/9] for everything
* Task 2: [1/9, ..., 1/9] for everything 
* Task 3: 
    * get average 'changed region' length and just pick the section in the middle
    * predict avg nr of 1s everywhere
* Task 4: do nothing

We should probably evaluate these for ourselves at least before releasing our baselines (which will look a bit silly if they don't win!). Any other dumb ones to propose?

**Comment by apmcleod** [comment]
- Task 3: Predict all 0's would probably be more accurate, similar to task 4 baseline.

**Comment by apmcleod** [comment]
I'll keep updating this comment with results.

Task 1:
- F-measure = ...
- Loss = `8/9 * CrossEnt([1/9, 8/9], 1) + 1/9 * CrossEnt([1/9, 8/9], 0)` = 0.4645
Task 2:
- Acc = 1/9
- Loss = `CrossEnt(np.ones(9) / 9, 0)` = 2.1972
Task 3 (Predict avg nr of 1s everywhere):
- F-measure = ...
- Loss = ...
Task 4 (Reproduce input):
- Helpfulness = `0.5 * 8/9 + 1 * 1/9` = 0.5556
- Loss = ...

**Comment by apmcleod** [comment]
The above numbers are in theory. The below are in practice:

Task 1:
-

**Comment by JamesOwers** [comment]
Either sort out maths of Task 4 dumb baseline, or rerun empirical on ACME1.0

**Comment by apmcleod** [comment]
Switched milestone because I have run on the current acme and that is in the submitted paper, but we should redo the "in practice" numbers with official ACME 1.0, when we have it.

**Comment by JamesOwers** [comment]
Maybe I'll make the maths work. @apmcleod - rerun baselines on real ACME dataset.

**Comment by apmcleod** [comment]
ACME v1.0 numbers:

Task 1-3 number gotten from adding `print(counts / np.sum(counts))` to `get_inverse_weights` and running each task with `--weight`. Should we make this doable without editing the code? Maybe not important.

Task 1:
[p(0) p(1)] = [0.11100006 0.88899994]

Task 2:
[p(0) p(1) ... p(8)] = [0.11100006 0.11105558 0.11127769 0.11138875 0.11111111 0.11094453
 0.10938975 0.11133322 0.11249931]

Task 3:
[p(0) p(1)] = [0.93126526 0.06873474]

Task 4:
TODO

**Comment by apmcleod** [comment]
I wrote a script for this. Will push to a "rule" branch. Output:

Task 1 loss = 0.4685294032096863
Task 1 rev F-measure = 0.0
Task 2 loss = 2.1972384452819824
Task 2 acc = 0.11639271434917814
Task 3 loss = 0.4128188192844391
Task 3 F-measure = 0.0
Task 4 loss = 0.6863518953323364
Task 4 Helpfulness = 0.558196357174589
